### PR TITLE
Modernize Rack::Protection::ContentSecurityPolicy

### PR DESCRIPTION
**Pull request by grempe**
Issue #1201 Modernizes Rack::Protection::ContentSecurityPolicy to support current CSP2 and CSP3 directives.

Issue #1200 Fixes a minor security warning on `bundle install` in a fresh repo.

Removes inline docs which were incomplete and insufficient for what is now
quite complex in CSP2 and CSP3. Adds additional current links to excellent
doc resources and tools.

### CODE REVIEW

**Comment by grempe** [comment]
The build failures above are unrelated to these commits.

From the build logs:

```
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.
...
An error occurred while installing mini_racer (0.1.7), and Bundler
```

**Comment by CGA1123** [comment]
Travis failing is related to #1199

**Comment by namusyaka** [comment]
This will be succeed because g++-4.8 is not available on ubuntu-trusty.

File: rack-protection/lib/rack/protection/content_security_policy.rb [hunk 1]
```diff
@@ -6,58 +6,73 @@ module Protection
...
-    # sandbox::     if the sandbox directive is present, the page will be 
-    #               treated as though it was loaded inside of an iframe with
-    #               a sandbox attribute.
     class ContentSecurityPolicy < Base
-      default_options :default_src => :none, :script_src => "'self'", :img_src => "'self'", :style_src => "'self'", :connect_src => "'self'", :report_only => false
+      default_options default_src: :none, script_src: "'self'",
+                      img_src: "'self'", style_src: "'self'",
+                      connect_src: "'self'", report_only: false
+
+      DIRECTIVES = [:base_uri, :child_src, :connect_src, :default_src,
```
Comment (esparta) [inline rack-protection/lib/rack/protection/content_security_policy.rb]: how about use %i(..) for the list directives ?

Comment (grempe) [inline rack-protection/lib/rack/protection/content_security_policy.rb]: Done, see d547b7f

**Pull request merged**